src/top/nn/nn.cc

```diff
@@ -82,8 +118,23 @@ NNVM_REGISTER_ELEMWISE_UNARY_OP(relu)
    max(input, 0)
 
 )code" NNVM_ADD_FILELINE)
+.set_attr<FGradient>(
+  "FGradient", [](const NodePtr& n,
+                  const std::vector<NodeEntry>& ograds) {
+    return MakeGradNode("_relu_grad", n,
+                        {ograds[0], n->inputs[0]});
+})
 .set_support_level(1);
 
+NNVM_REGISTER_OP(_relu_grad)
```

Comment:
not register gradient for simple operators like relu, instead simply return a composed graph containing the gradient